Adding prisma to nestjs project should note/flag/modify `start:prod` script

### Bug description

To build a production version of a nestjs project (**_without prisma_**), you run

`yarn build`

Which creates a dist directory, with a main.js file that can be run by:

`yarn start:prod`

which just executes:

`node dist/main`

When **adding `prisma`** to a `nestjs` project, something in `prisma` modifies the build such that instead of `./dist/` having the main.js file, it now has another level of indirection (as well as the new `prisma` directory in ` so it looks like:

```
prisma/
src/
src/main.js 
tsconfig.build.tsbuildinfo 
```

Now when **running the production build**, the `package.json` `start:prod` script, the script reports:
`Error: Cannot find module '/tmp/rest-nestjs/dist/main'`

As it still references `dist/main(.js)` instead of the new `dist/src/main(.js)`

Unfortunately, I haven't had the time yet to dig into how we get here, so maybe the first question is, what changes the build?

- installation of `@prisma/client`?
- some hook into nests build? 
- `prisma generate`?
- something else?

I presume the correct fix would be to modify the prod:start if it is dist/main, or if it has changed (or doesn't exist) throw an error?
Also maybe add a note/flag saying it was modified?

And as I was tracking this down, I was (briefly) trying to figure out in the docs where and what exactly was happening here. Did I miss the explanations to what is happening here? I can be obtuse sometimes. ;-)

Thanks!



### How to reproduce

Follow instructions from [here](https://github.com/prisma/prisma-examples/tree/latest/typescript/rest-nestjs):

```
 curl https://codeload.github.com/prisma/prisma-examples/tar.gz/latest | tar -xz --strip=2 prisma-examples-latest/typescript/rest-nestjs
cd rest-nestjs
yarn install
...
yarn build
ls -l dist/
yarn start:prod
Error: Cannot find module '/tmp/rest-nestjs/dist/main'

cat package.json| grep prod
    "start:prod": "node dist/main",
```
Instead run:
```
node dist/src/main
```

And all works fine


### Expected behavior

The app runs fine

### Prisma information

Uses `prisma-examples-latest/typescript/rest-nestjs`

### Environment & setup

- OS: WSL (Windows 10)
- Database: PostgreSQL (but N/A)
- Node.js version:  v16.15.0


### Prisma Version

```
prisma                  : 3.13.0
@prisma/client          : 3.13.0
Current platform        : debian-openssl-1.1.x
Query Engine (Node-API) : libquery-engine efdf9b1183dddfd4258cd181a72125755215ab7b (at node_modules/@prisma/engines/libquery_engine-debian-openssl-1.1.x.so.node)
Migration Engine        : migration-engine-cli efdf9b1183dddfd4258cd181a72125755215ab7b (at node_modules/@prisma/engines/migration-engine-debian-openssl-1.1.x)
Introspection Engine    : introspection-core efdf9b1183dddfd4258cd181a72125755215ab7b (at node_modules/@prisma/engines/introspection-engine-debian-openssl-1.1.x)
Format Binary           : prisma-fmt efdf9b1183dddfd4258cd181a72125755215ab7b (at node_modules/@prisma/engines/prisma-fmt-debian-openssl-1.1.x)
Default Engines Hash    : efdf9b1183dddfd4258cd181a72125755215ab7b
Studio                  : 0.459.0
```

--- a/typescript/graphql-nestjs-sdl-first/package.json
+++ b/typescript/graphql-nestjs-sdl-first/package.json
@@ -12,7 +12,7 @@
     "start": "nest start",
     "dev": "nest start --watch",
     "start:debug": "nest start --debug --watch",
-    "start:prod": "node dist/main",
+    "start:prod": "node dist/src/main",
     "lint": "eslint \"{src,apps,libs,test}/**/*.ts\" --fix",
     "test": "jest",
     "test:watch": "jest --watch",

--- a/typescript/graphql-nestjs/package.json
+++ b/typescript/graphql-nestjs/package.json
@@ -12,7 +12,7 @@
     "start": "nest start",
     "dev": "nest start --watch",
     "start:debug": "nest start --debug --watch",
-    "start:prod": "node dist/main",
+    "start:prod": "node dist/src/main",
     "lint": "eslint \"{src,apps,libs,test}/**/*.ts\" --fix",
     "test": "jest",
     "test:watch": "jest --watch",

--- a/typescript/rest-nestjs/package.json
+++ b/typescript/rest-nestjs/package.json
@@ -12,7 +12,7 @@
     "start": "nest start",
     "dev": "nest start --watch",
     "start:debug": "nest start --debug --watch",
-    "start:prod": "node dist/main",
+    "start:prod": "node dist/src/main",
     "lint": "eslint \"{src,apps,libs,test}/**/*.ts\" --fix",
     "test": "jest",
     "test:watch": "jest --watch",

